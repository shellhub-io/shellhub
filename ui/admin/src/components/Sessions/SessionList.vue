<template>
  <div>
    <DataTable
      :headers="headers"
      :items="sessions"
      :itemsPerPage="itemsPerPage"
      :nextPage="next"
      :previousPage="prev"
      :loading="loading"
      :totalCount="numberSessions"
      :page="page"
      :actualPage="page"
      @changeItemsPerPage="changeItemsPerPage"
      @clickNextPage="next"
      @clickPreviousPage="prev"
      data-test="session-list"
    >
      <template v-slot:rows>
        <tr v-for="(session, index) in sessions" :key="index">
          <td>
            <v-icon v-if="session.active" color="success">
              mdi-check-circle
            </v-icon>
            <v-tooltip anchor="bottom" v-else>
              <template v-slot:activator="{ props }">
                <v-icon v-bind="props"> mdi-check-circle </v-icon>
              </template>
              <span>{{ lastSeen(session.last_seen) }}</span>
            </v-tooltip>
          </td>
          <td>
            <v-chip>
              {{ displayOnlyTenCharacters(session.uid) }}
              <v-tooltip activator="parent" anchor="bottom">{{
                session.uid
              }}</v-tooltip>
            </v-chip>
          </td>
          <td>
            <span
              @click="redirectToDevice(session.device.uid)"
              @keypress.enter="redirectToDevice(session.device.uid)"
              tabindex="0"
              class="hover"
            >
              {{ session.device.name }}
            </span>
          </td>
          <td>
            {{ session.username }}
          </td>
          <td class="d-flex justify-center align-center">
            <v-tooltip anchor="bottom" v-if="session.authenticated">
              <template v-slot:activator="{ props }">
                <v-icon v-bind="props">mdi-shield-check </v-icon>
              </template>
              <span>User has been authenticated</span>
            </v-tooltip>
            <v-tooltip anchor="bottom" v-else>
              <template v-slot:activator="{ props }">
                <v-icon v-bind="props" color="error"> mdi-shield-alert </v-icon>
              </template>
              <span>User has not been authenticated</span>
            </v-tooltip>
          </td>
          <td>
            <v-chip density="compact" label>{{ session.ip_address }}</v-chip>
          </td>
          <td>
            <span>{{ formatDate(session.started_at) }}</span>
          </td>
          <td>
            {{ formatDate(session.last_seen) }}
          </td>
          <td>
            <v-tooltip bottom anchor="bottom">
              <template v-slot:activator="{ props }">
                <v-icon
                  tag="a"
                  dark
                  v-bind="props"
                  @click="goToSession(session.uid)"
                  @keypress.enter="goToSession(session.tenant_id)"
                  tabindex="0"
                >mdi-information
                </v-icon>
              </template>
              <span>Details</span>
            </v-tooltip>
          </td>
        </tr>
      </template>
    </DataTable>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, onMounted, ref, watch } from "vue";
import { useRouter } from "vue-router";
import { useStore } from "../../store";
import DataTable from "../DataTable.vue";
import { lastSeen, formatDate } from "../../hooks/formateDate";
import displayOnlyTenCharacters from "../../hooks/string";
import { INotificationsError } from "../../interfaces/INotifications";

export default defineComponent({
  setup() {
    const store = useStore();
    const router = useRouter();
    const itemsPerPage = ref(10);
    const loading = ref(false);
    const page = ref(1);

    const getSessions = async (perPagaeValue: number, pageValue: number) => {
      try {
        loading.value = true;

        const hasSessions = await store.dispatch("sessions/fetch", {
          perPage: perPagaeValue,
          page: pageValue,
        });

        if (!hasSessions) {
          page.value--;
        }

        loading.value = false;
      } catch (error) {
        store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.sessionList);
      }
    };

    onMounted(async () => {
      try {
        loading.value = true;
        getSessions(itemsPerPage.value, page.value);
      } catch (error) {
        store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.sessionList);
      } finally {
        loading.value = false;
      }
    });

    const next = async () => {
      await getSessions(itemsPerPage.value, ++page.value);
    };

    const prev = async () => {
      try {
        if (page.value > 1) await getSessions(itemsPerPage.value, --page.value);
      } catch (error) {
        store.dispatch("snackbar/setSnackbarErrorDefault");
      }
    };

    const changeItemsPerPage = async (newItemsPerPage: number) => {
      itemsPerPage.value = newItemsPerPage;
    };

    watch(itemsPerPage, async () => {
      await getSessions(itemsPerPage.value, page.value);
    });

    const sessions = computed(() => store.getters["sessions/sessions"]);
    const numberSessions = computed(() => store.getters["sessions/numberSessions"]);

    const redirectToDevice = (deviceId: string) => {
      router.push({ name: "deviceDetails", params: { id: deviceId } });
    };

    const goToSession = (sessionId: string) => {
      router.push({ name: "sessionDetails", params: { id: sessionId } });
    };

    return {
      headers: [
        {
          text: "Active",
          value: "active",
        },
        {
          text: "Id",
          value: "uid",
        },
        {
          text: "Device",
          value: "device",
        },
        {
          text: "Username",
          value: "username",
        },
        {
          text: "Authenticated",
          value: "authenticated",
        },
        {
          text: "IP Address",
          value: "ip_address",
        },
        {
          text: "Started",
          value: "started_at",
        },
        {
          text: "Last Seen",
          value: "last_seen",
        },
        {
          text: "Actions",
          value: "actions",
        },
      ],
      itemsPerPage,
      loading,
      page,
      sessions,
      numberSessions,
      lastSeen,
      displayOnlyTenCharacters,
      redirectToDevice,
      formatDate,
      goToSession,
      changeItemsPerPage,
      next,
      prev,
    };
  },
  components: { DataTable },
});
</script>

<style scoped>
.hover {
  cursor: pointer;
  text-decoration: underline;
}
</style>
