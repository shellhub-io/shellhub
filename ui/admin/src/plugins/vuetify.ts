// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
