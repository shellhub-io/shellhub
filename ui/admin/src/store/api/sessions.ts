import { adminApi } from "../../api/http";

const fetchSessions = async (perPage: number, page: number) => adminApi.getSessionsAdmin(page, perPage);

const getSession = async (uid: string) => adminApi.getSessionAdmin(uid);

export { fetchSessions, getSession };
