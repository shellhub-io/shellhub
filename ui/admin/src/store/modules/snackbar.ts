import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
