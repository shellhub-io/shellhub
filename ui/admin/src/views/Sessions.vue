<template>
  <div>
    <h1>Sessions</h1>
  </div>
  <v-card class="mt-2">
    <SessionList />
  </v-card>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import SessionList from "../components/Sessions/SessionList.vue";

export default defineComponent({ components: { SessionList } });
</script>
