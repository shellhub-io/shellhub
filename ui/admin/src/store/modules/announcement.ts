// stores/announcement.ts
import { defineStore } from "pinia";
import { Announcement } from "@admin/api/client/api";
import {
  postAnnouncement,
  updateAnnouncement,
  deleteAnnouncement,
  getAnnouncement,
  getListAnnouncements,
} from "../api/announcement";

interface AnnouncementState {
  announcements: Array<Announcement>;
  announcement: Announcement;
  numberAnnouncements: number;
  page: number;
  perPage: number;
  orderBy: "asc" | "desc";
}

export const useAnnouncementStore = defineStore("announcement", {
  state: (): AnnouncementState => ({
    announcements: [],
    announcement: {} as Announcement,
    numberAnnouncements: 0,
    page: 1,
    perPage: 10,
    orderBy: "asc",
  }),

  getters: {
    getAnnouncements: (state) => state.announcements,
    getAnnouncement: (state) => state.announcement,
    getNumberAnnouncements: (state) => state.numberAnnouncements,
    getPage: (state) => state.page,
    getPerPage: (state) => state.perPage,
    getOrderBy: (state) => state.orderBy,
  },

  actions: {
    async postAnnouncement(announcement: Announcement) {
      try {
        const { data } = await postAnnouncement(announcement as Required<Announcement>);
        this.announcement = data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },

    async updateAnnouncement(uuid: string, announcement: Announcement) {
      try {
        const { data } = await updateAnnouncement(uuid, announcement as Required<Announcement>);
        this.announcement = data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },

    async fetchAnnouncement(uuid: string) {
      try {
        const { data } = await getAnnouncement(uuid);
        this.announcement = data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },

    async fetchAnnouncements({ page, perPage, orderBy }: { page: number; perPage: number; orderBy: "asc" | "desc" }) {
      try {
        const res = await getListAnnouncements(page, perPage, orderBy);
        if (res.data && res.data.length) {
          this.announcements = res.data;
          this.numberAnnouncements = parseInt(res.headers["x-total-count"], 10);
          return res;
        }
        return false;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },

    async deleteAnnouncement(uuid: string) {
      try {
        const { data } = await deleteAnnouncement(uuid);
        this.announcement = data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },

    setPageAndPerPage({ page, perPage }: { page: number; perPage: number }) {
      this.page = page;
      this.perPage = perPage;
    },

    setOrderBy(orderBy: "asc" | "desc") {
      this.orderBy = orderBy;
    },

    clearAnnouncements() {
      this.announcements = [];
    },

    clearAnnouncement() {
      this.announcement = {} as Required<Announcement>;
    },
  },
});

export default useAnnouncementStore;
