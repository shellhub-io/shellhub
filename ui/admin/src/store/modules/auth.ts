// stores/auth.ts
import { defineStore } from "pinia";
import { login, getToken } from "../api/auth";

export interface AuthState {
  status: string;
  token: string;
  user: string;
}

export const useAuthStore = defineStore("auth", {
  state: (): AuthState => ({
    status: "",
    token: localStorage.getItem("cloud_token") || "",
    user: localStorage.getItem("cloud_user") || "",
  }),

  getters: {
    isLoggedIn: (state) => !!state.token,
    currentUser: (state) => state.user,
    authStatus: (state) => state.status,
  },

  actions: {
    async login(user: { username: string; password: string }) {
      this.status = "loading";

      try {
        const { data } = await login(user);
        this.status = "success";
        this.token = data.token ?? "";
        this.user = data.user ?? "";

        localStorage.setItem("cloud_token", data.token || "");
        localStorage.setItem("cloud_user", data.user || "");
      } catch (error) {
        this.status = "error";
        throw error;
      }
    },

    async loginToken(user: { id: string }) {
      try {
        const resp = await getToken(user.id);
        return resp.data.token;
      } catch (error) {
        this.status = "error";
        throw error;
      }
    },

    logout() {
      this.status = "";
      this.token = "";
      this.user = "";

      localStorage.removeItem("cloud_token");
      localStorage.removeItem("cloud_user");
    },
  },
});

export default useAuthStore;
