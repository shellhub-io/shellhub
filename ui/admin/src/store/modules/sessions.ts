import { defineStore } from "pinia";
import { Session } from "@admin/api/client";
import * as apiSession from "../api/sessions";

export const useSessionsStore = defineStore("sessions", {
  state: () => ({
    sessions: [] as Array<Session>,
    session: {} as Session,
    numberSessions: 0,
  }),

  getters: {
    getSessions: (state) => state.sessions,
    getSession: (state) => state.session,
    getNumberSessions: (state) => state.numberSessions,
  },

  actions: {
    async fetch(data: { perPage: number; page: number }) {
      const res = await apiSession.fetchSessions(data.perPage, data.page);

      if (res.data.length) {
        this.sessions = res.data;
        this.numberSessions = parseInt(res.headers["x-total-count"], 10);
        return res;
      }

      return false;
    },

    async get(uid: string) {
      const res = await apiSession.getSession(uid);
      this.session = res.data;
    },

    clearListSessions() {
      this.sessions = [];
      this.numberSessions = 0;
    },

    clearObjectSession() {
      this.session = {} as Session;
    },
  },
});

export default useSessionsStore;
