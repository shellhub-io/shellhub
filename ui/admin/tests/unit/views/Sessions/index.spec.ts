import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { createPinia, setActivePinia } from "pinia";
import { useSessionsStore } from "@admin/store/modules/sessions";
import { SnackbarPlugin } from "@/plugins/snackbar";
import routes from "../../../../src/router";
import Sessions from "../../../../src/views/Sessions.vue";

type SessionsWrapper = VueWrapper<InstanceType<typeof Sessions>>;

describe("Sessions", () => {
  let wrapper: SessionsWrapper;

  beforeEach(() => {
    const pinia = createPinia();
    setActivePinia(pinia);

    const sessionsStore = useSessionsStore();
    sessionsStore.fetch = vi.fn();

    const vuetify = createVuetify();

    wrapper = mount(Sessions, {
      global: {
        plugins: [pinia, vuetify, routes, SnackbarPlugin],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should render all the components in the screen", () => {
    expect(wrapper.find("h1").text()).toContain("Sessions");
    expect(wrapper.find("[data-test='session-list']").exists()).toBe(true);
  });
});
