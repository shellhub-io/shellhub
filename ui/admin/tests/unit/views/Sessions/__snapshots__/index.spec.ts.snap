// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sessions > Renders the component 1`] = `
"<div>
  <h1>Sessions</h1>
</div>
<div class="v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-d900b695="">
    <div data-v-d900b695="" data-test="session-list">
      <div class="v-table v-theme--light v-table--density-default bg-background border rounded text-center">
        <!---->
        <div class="v-table__wrapper">
          <table>
            <thead class="bg-v-theme-background">
              <tr>
                <th class="text-center"><span>Active</span></th>
                <th class="text-center"><span>Id</span></th>
                <th class="text-center"><span>Device</span></th>
                <th class="text-center"><span>Username</span></th>
                <th class="text-center"><span>Authenticated</span></th>
                <th class="text-center"><span>IP Address</span></th>
                <th class="text-center"><span>Started</span></th>
                <th class="text-center"><span>Last Seen</span></th>
                <th class="text-center"><span>Actions</span></th>
              </tr>
            </thead>
            <tbody class="pa-4 text-subtitle-2">
              <tr>
                <td colspan="9" class="pa-4 text-subtitle-2 text-center"> No data available </td>
              </tr>
            </tbody>
          </table>
        </div>
        <!---->
      </div>
      <!--v-if-->
      <!--v-if-->
      <div class="d-flex w-100 justify-end align-center"><span class="text-subtitle-2 mr-4">Items per page:</span>
        <div>
          <div class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-2">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!---->
                  <!---->
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!---->
                    <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" type="text" id="input-v-0" aria-describedby="input-v-0-messages" outlined="" value="10">
                  </div>
                  <!---->
                </div>
                <!---->
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
        </div>
        <div class="d-flex align-center"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button><span class="text-subtitle-2">1 of 1</span><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button></div>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
